dali/base/dafdesc.cpp

```diff
@@ -2934,30 +2937,33 @@ bool setReplicateDir(const char *dir,StringBuffer &out,bool isrep,const char *ba
     unsigned count = 0;
     unsigned i;
     for (i=0;d[i]&&dir[i]&&(d[i]==dir[i]);i++)
-        if (isPathSepChar(dir[i])) {
+    {
+        if (isPathSepChar(dir[i]))
+        {
             match = i;
             count++;
         }
+    }
     const char *r = repDir?repDir:queryBaseDirectory(grp_unknown, isrep ? 1 : 0,os);
-    if (d[i]==0) {
-        if ((dir[i]==0)||isPathSepChar(dir[i])) {
+    if (d[i]==0)
+    {
+        if (dir[i]==0)
```

Comment:
Worth adding a comment to clarify the cases.  E.g.,
//directory is an exact match for the current base, so return replica unmodified.